docs/architecture/stack/evm-state-manager/index.mdx

```diff
@@ -1,31 +1,251 @@
+            A[root] --> B[node] & C[node]
+            B --> D[node] & E[node]
+            C --> F[node] & G[node]
+    ```
+  </div>
+</div>
+
+Each node, regardless of type, is represented by a cryptographic hash which encodes data about its 
+properties — for example, the contents of your account. Each hash encodes the hashes of its child 
+nodes. Taken to its full extent, this cascading system means the root encodes data of the state of 
+every single account on the blockchain. 
+
+Cryptographic hashes are deterministic, which means you can reverse the hash function to get the 
+data which it encoded. If you have the hash of the root—the only node without a parent—you can 
+theoretically derive from it the data of any node in the entire tree. 
+
+As a layer 2 (L2) network, Linea is in the business of making transacting faster and more efficient.
+Viewed from this angle, the standard implementation of a Merkle tree, described above, can present
+problems. 
+
+A Merkle tree must be computed anew for every single block, accounting for any state changes 
+resulting from the transactions included in the block. This can quickly snowball to the point where 
+an incredible amount of computation is required to finalize a block. 
+
+### Sparse Merkle trees
+
+Linea's state management uses sparse Merkle trees to minimize computation and contribute to the 
+blockchain's efficiency. 
+
+A sparse Merkle tree is a variation of a standard Merkle tree where not all leaf nodes are filled 
+with data; instead, data is only stored in nodes where it's needed. It is a complete tree of fixed 
+depth, meaning that all branches of the tree have the same length—i.e. the same number of leaves. 
+
+At initialization, all leaf nodes are set to a default value, which is typically a hash of a 
+specific value, such as zero. Because all leaf nodes have the same hash value, the parent nodes and 
+higher-level nodes also have the same hash value. A node whose hash is the default value for its 
+level is therefore considered to represent an _empty_ subtree. 
+
+<div class="center-container">
+  <div class="mermaid-medium">
+    ```mermaid
+    flowchart TD
+        A[root] --> B["`**node A**`"
+        null] & C["`**node B**`"
+        contains data]
+        B --> D["`**node C**`":
+        null] & E["`**node D:**`"
+        null]
+        C --> F["`**node E**`":
+        data] & G["`**node F**`":
+        data]
+    ```
+  </div>
+</div>
+
+In the example above, the children of node A (leaves) contain null values, which means node A does 
+too. Node B, meanwhile, reflects that its children also contain values.
+
+With this construction, we do not need to keep track of every individual node's hash. Instead, we 
+can assume hashes that reflect the default value are empty, and the subtree or node that lies 
+further down the chain of nodes can be disregarded; we only need to pay attention to the ones that 
+correspond to _non-empty_ subtrees.
+
+## Cryptographic accumulator
```

Comment:
I'm a bit confused about the relationship between the Cryptographic accumulator and SMT.
Is the SMT being discussed within the context of the Cryptographic accumulator topic?

Not sure the topic structure helps showing the relationship.

Reply:
Rearranged the paragraphs and added a sentence or two to make the flow into this section a bit better